### DISCUSSION TITLE
As soon as enable controlnet on deforum, I can't generate any videos

### DISCUSSION THREAD

**Discussion by Maxa1577** [Mikubill/sd-webui-controlnet#550] [General]
I get this when generating
TypeError: Object of type _TemporaryFileWrapper is not JSON serializable

Anyone got a solution? I've rolled back the extension, reinstalled everything but I can't get it working.

`Deforum extension for auto1111 webui, v2.2b
Git commit: dbb852f3 (Thu Mar  9 14:17:49 2023)
Unpacking ControlNet base video
Exporting Video Frames (1 every 1) frames to C:\stable-diffusion\AUTO_INSTALL\stable-diffusion-webui\outputs/img2img-images\Deforum\controlnet_inputframes...
Trying to extract frames from video with input FPS of 23.976023976023978. Please wait patiently.
Successfully extracted 634 frames from video.
Loading 120 input frames from C:\stable-diffusion\AUTO_INSTALL\stable-diffusion-webui\outputs/img2img-images\Deforum\controlnet_inputframes and saving video frames to C:\stable-diffusion\AUTO_INSTALL\stable-diffusion-webui\outputs/img2img-images\Deforum
ControlNet base video unpacked!
Saving animation frames to:
C:\stable-diffusion\AUTO_INSTALL\stable-diffusion-webui\outputs/img2img-images\Deforum
Error completing request
Arguments: ('task(59qsmyaqb8yyi8t)', None, False, '', '2D', 120, 'replicate', '0:(0)', '0:(1.0025+0.002*sin(1.25*3.14*t/30))', '0:(0)', '0:(0)', '0:(1.75)', '0:(0.5)', '0:(0.5)', '0:(0)', '0:(0)', '0:(0)', False, '0:(0)', '0:(0)', '0:(0)', '0:(53)', '0: (0.065)', '0: (0.65)', '0: (1.0)', '0: (7)', '0:(1.5)', False, '0:(1)', '0:(0)', False, '0: (25)', '0: (70)', '0: (1)', '0: (200)', '0: (10000)', '0:(5), 1:(-1), 219:(-1), 220:(5)', False, '0: ("Euler a")', '0: ("!({everywhere}^({init_mask}|{video_mask}) ) ")', False, '0: ("!({everywhere}^({init_mask}|{video_mask}) ) ")', False, '0: ("model1.ckpt"), 100: ("model2.ckpt")', False, '0: (2)', '0: (5)', '0: (1.0)', '0: (0.35)', '0: (0.0)', 'Match Frame 0 LAB', 1.0, False, '2', 'perlin', 8, 8, 4, 0.5, True, 0.2, 'border', 'bicubic', False, 'https://github.com/hithereai/d/releases/download/m/vid.mp4', 1, 0, -1, False, False, '/content/video_in.mp4', False, '20230129210106', '{\n    "0": "tiny cute swamp bunny, highly detailed, intricate, ultra hd, sharp photo, crepuscular rays, in focus, by tomasz alen kopera",\n    "30": "anthropomorphic clean cat, surrounded by fractals, epic angle and pose, symmetrical, 3d, depth of field, ruan jia and fenghua zhong",\n    "60": "a beautiful coconut --neg photo, realistic",\n    "90": "a beautiful durian, trending on Artstation"\n}\n    ', '', '', 512, 512, False, False, -1, 'Euler a', False, 0, 0, 25, 0.0, 1, True, True, False, False, False, 'Deforum', '{timestring}_{index}_{prompt}.png', 'iter', 1, False, False, True, 0, 'https://github.com/hithereai/d/releases/download/m/kaba.png', False, False, False, True, 'https://github.com/hithereai/d/releases/download/m/mask.jpg', 1.0, 1.0, 4, 1, True, 4, 'reroll', False, 15, False, 'FFMPEG mp4', 'C:\\stable-diffusion\\AUTO_INSTALL\\stable-diffusion-webui\\venv\\lib\\site-packages\\imageio_ffmpeg\\binaries\\ffmpeg-win64-v4.2.2.exe', '17', 'slow', 'None', 'https://freetestdata.com/wp-content/uploads/2021/09/Free_Test_Data_1MB_MP3.mp3', False, 'realesr-animevideov3', 'x2', True, False, 'x0_pred', 'C:/SD/20230124234916_%09d.png', 'testvidmanualsettings.mp4', False, 'None', 2, False, 2, False, '', True, False, 'None', True, 'None', False, 'Farneback', False, 'None', False, 'None', False, False, '0:(1)', '0:(0.5)', '0:(1)', '0:(100)', '0:(0)', False, '{\n    "0": "https://user-images.githubusercontent.com/121192995/215279228-1673df8a-f919-4380-b04c-19379b2041ff.png",\n    "50": "https://user-images.githubusercontent.com/121192995/215279281-7989fd6f-4b9b-4d90-9887-b7960edd59f8.png",\n    "100": "https://user-images.githubusercontent.com/121192995/215279284-afc14543-d220-4142-bbf4-503776ca2b8b.png",\n    "150": "https://user-images.githubusercontent.com/121192995/215279286-23378635-85b3-4457-b248-23e62c048049.jpg",\n    "200": "https://user-images.githubusercontent.com/121192995/215279228-1673df8a-f919-4380-b04c-19379b2041ff.png"\n}', '0:(0.75)', '0:(0.35)', '0:(0.25)', '0:(20)', '0:(0.075)', <tempfile._TemporaryFileWrapper object at 0x0000028150251B70>, None, True, False, False, False, 'none', 'None', 1, 1, 64, 64, 64, 'Scale to Fit (Inner Fit)') {}Traceback (most recent call last):
  File "C:\stable-diffusion\AUTO_INSTALL\stable-diffusion-webui\modules\call_queue.py", line 56, in f
    res = list(func(*args, **kwargs))
  File "C:\stable-diffusion\AUTO_INSTALL\stable-diffusion-webui\modules\call_queue.py", line 37, in f
    res = func(*args, **kwargs)
  File "C:\stable-diffusion\AUTO_INSTALL\stable-diffusion-webui\extensions\deforum-for-automatic1111-webui\scripts\deforum.py", line 84, in run_deforum
    render_animation(args, anim_args, video_args, parseq_args, loop_args, controlnet_args, root.animation_prompts, root)  File "C:\stable-diffusion\AUTO_INSTALL\stable-diffusion-webui/extensions/deforum-for-automatic1111-webui/scripts\deforum_helpers\render.py", line 69, in render_animation
    save_settings_from_animation_run(args, anim_args, parseq_args, loop_args, controlnet_args, video_args, root)
  File "C:\stable-diffusion\AUTO_INSTALL\stable-diffusion-webui/extensions/deforum-for-automatic1111-webui/scripts\deforum_helpers\settings.py", line 60, in save_settings_from_animation_run
    json.dump(s, f, ensure_ascii=False, indent=4)
  File "C:\Users\IT\AppData\Local\Programs\Python\Python310\lib\json\__init__.py", line 179, in dump
    for chunk in iterable:
  File "C:\Users\IT\AppData\Local\Programs\Python\Python310\lib\json\encoder.py", line 431, in _iterencode
    yield from _iterencode_dict(o, _current_indent_level)
  File "C:\Users\IT\AppData\Local\Programs\Python\Python310\lib\json\encoder.py", line 405, in _iterencode_dict
    yield from chunks
  File "C:\Users\IT\AppData\Local\Programs\Python\Python310\lib\json\encoder.py", line 438, in _iterencode
    o = _default(o)
  File "C:\Users\IT\AppData\Local\Programs\Python\Python310\lib\json\encoder.py", line 179, in default
    raise TypeError(f'Object of type {o.__class__.__name__} '
TypeError: Object of type _TemporaryFileWrapper is not JSON serializable`

**Comment by hectorruizlaph** [comment]
did you found the solution?
